### DISCUSSION TITLE
GraphQL createCommitOnBranch is unverified - How do I make it verified?

### DISCUSSION THREAD

**Discussion by rashelle-hopkins-cb** [community/community#44009] [Enterprise] [answered]
### Select Topic Area

Question

### Body

Hello, I am creating a GitHub app in Enterprise. My app utilizes the GitHub GraphQL and REST API to create a commit and PR.

According to the GitHub docs, creating a commit with the `createCommitOnBranch` mutation: ["[is] automatically signed by GitHub if supported and will be marked as verified in the user interface."](https://docs.github.com/en/graphql/reference/mutations#commit-signing)

Using the [GitHub app web application flow](https://docs.github.com/en/developers/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps#web-application-flow), users are successfully authenticated and a PR is created on their behalf. However, the commit does not appear as verified. My own manually created commits appear as verified, but not the ones made through the GitHub app.

I searched the docs and general internet and can't find what the meaning of "if supported" from the first link means. I do not see any setting for enabling GitHub app verification. I also tried enabling GPG key permissions, among others, and commit signing still does not work. I can only find articles that say that this should work automatically.

My query looks like so, with the capitalized text replaced by actual values:
```
{
  "query": "mutation ($input: CreateCommitOnBranchInput!) {createCommitOnBranch(input: $input) { commit { url } }}",
  "variables": {
    "input": {
      "branch": {
        "repositoryNameWithOwner": "REPOOWNER/REPONAME",
        "branchName": "BRANCHNAME"
      },
      "expectedHeadOid": "OID",
      "message": {"headline": "MESSAGE"},
      "fileChanges": {
        "additions": [{
          "path": "PATH.yml",
          "contents": "CONTENTS"
        }]
      }
    }
  }
}
```

If anyone can assist me with ensuring commits are verified, or can help lead me in the right direction, I would greatly appreciate that. Thank you.

**Comment by mcintyre321** [comment, marked as answer]
Did you find an answer to this?

I wonder if [Web Commit Signing](https://docs.github.com/en/enterprise-server@3.5/admin/configuration/configuring-your-enterprise/configuring-web-commit-signing) needs to be enabled.

**Reply by rashelle-hopkins-cb** [reply]
Yes, turns out that was the case. Thank you!
